### PR TITLE
Add scripts for downloading and restoring backups

### PR DESCRIPTION
**Pull request by giorgiosironi**
For https://github.com/elifesciences/issues/issues/4607

Defaults to 1 day back, using /ext/tmp

Assumes AWS credentials are available, I plan for Jenkins to pass some temporary credentials in through environment variables.

### CODE REVIEW

File: scripts/download-backup.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+set -e
+# Usage: TMP=FOLDER scripts/donload-backup.sh DELAY_IN_DAYS
```
Comment (nlisgo) [inline scripts/download-backup.sh]: nitpick: `scripts/donload-back.sh` should be `scripts/download-backup.sh`

File: example.env [hunk 2]
```diff
@@ -0,0 +1 @@
+export A=1 B=2
```
Comment (nlisgo) [inline example.env]: Was this an intentional addition?

File: scripts/download-backup.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+set -e
+# Usage: TMP=FOLDER scripts/download-backup.sh DELAY_IN_DAYS
+# Example: TMP=/ext/tmp scripts/download-backup.sh 1
+
+# arguments and environment variables to influence behavior
+delay="${1:-1}" # default to 1 day back
+TMP="${TMP:-/ext/tmp/}"
+
+backupLocation='s3://elife-app-backups/journal-cms/'
+selectedMonth="$(date -d "-$delay days" +%Y%m)"
```
Comment (giorgiosironi) [inline scripts/download-backup.sh]: this must be made portable across Linux/Mac

Comment (nlisgo) [reply scripts/download-backup.sh]: I amended this locally to work for me: `selectedMonth="$(date -v-$(delay)d +%Y%m)"`

The above solution would not work on Linux.

File: example.env (hunk 2 above)
Comment (giorgiosironi) [reply example.env]: Will remove

**Comment by giorgiosironi** [comment]
Added another script, which I am testing now.

**Comment by giorgiosironi** [comment]
Seems to have worked:
```
elife@continuumtest--journal-cms:/srv/journal-cms$ curl -s localhost/people | jq .total
445
```

File: scripts/restore-backup.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+databaseArchive="$(ls "$TMP/*-elife_2_0-mysql.gz")"
+
+echo "Extracting $filesArchive"
+tar -xzf "$filesArchive" -C /ext/tmp
+
+echo "Restoring $filesArchive"
+rm -rf /srv/journal-cms/web/sites/default/files
+mv /ext/tmp/srv/journal-cms/web/sites/default/files web/sites/default/
+sudo chown -R www-data:www-data /srv/journal-cms/web/sites/default/files
+rm -rf /ext/tmp/srv
```
Comment (giorgiosironi) [inline scripts/restore-backup.sh]: need to also remove `$filesArchive`

**Comment by giorgiosironi** [comment]
Ready for new review, of the new script.

File: scripts/download-backup.sh [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+# Example: TMP=/ext/tmp scripts/download-backup.sh 1
+
+# arguments and environment variables to influence behavior
+delay="${1:-1}" # default to 1 day back
+TMP="${TMP:-/ext/tmp/}"
+
+backupLocation='s3://elife-app-backups/journal-cms/'
+if [ "$(uname -s)" == "Darwin" ]; then
+    # Mac OS X
+    selectedMonth="$(date "-v -${delay}d" +%Y%m)"
```
Comment (nlisgo) [inline scripts/download-backup.sh]: Removing the space in `"-v -${delay}d"` works for me. `"-v-${delay}d"`

File: scripts/restore-backup.sh [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+
+TMP="${TMP:-/ext/tmp}"
+# cannot use double quotes to allow bash to resolve * wildcards
+# shellcheck disable=SC2086
+filesArchive=$(find $TMP/*-archive-*.tar.gz)
+# shellcheck disable=SC2086
+databaseArchive=$(find $TMP/*-elife_2_0-mysql.gz)
+
+echo "Extracting $filesArchive"
+tar -xzf "$filesArchive" -C /ext/tmp
```
Comment (nlisgo) [inline scripts/restore-backup.sh]: Should this be:

```
tar -xzf "$filesArchive" -C $TMP
```
?

File: scripts/restore-backup.sh [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+echo "Restoring $filesArchive"
+rm -rf /srv/journal-cms/web/sites/default/files
```
Comment (nlisgo) [inline scripts/restore-backup.sh]: in order for this to work locally we should allow the web root to be overridden:

```
WEB="${WEB:-/srv/journal-cms/web}"
```

```
rm -rf $WEB/sites/default/files
mv $TMP/srv/journal-cms/web/sites/default/files $WEB/web/sites/default/
sudo chown -R www-data:www-data $WEB/sites/default/files
rm -rf $TMP/srv
```

File: scripts/restore-backup.sh [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+mv /ext/tmp/srv/journal-cms/web/sites/default/files web/sites/default/
+sudo chown -R www-data:www-data /srv/journal-cms/web/sites/default/files
+rm -rf /ext/tmp/srv
+rm -f "$filesArchive"
+
+echo "Restoring $databaseArchive"
+cd web
```
Comment (nlisgo) [inline scripts/restore-backup.sh]:
```
cd $WEB
```

Comment (giorgiosironi) [reply scripts/restore-backup.sh]: Has been substituted with your `ROOT` environment variable suggestion

File: scripts/restore-backup.sh (hunk 7 above)
Comment (giorgiosironi) [reply scripts/restore-backup.sh]: `ROOT` is now allowed

File: scripts/restore-backup.sh (hunk 6 above)
Comment (giorgiosironi) [reply scripts/restore-backup.sh]: `TMP` has been substituted here

**Comment by giorgiosironi** [comment]
All suggestions incorporated, merging on green.

**Comment by giorgiosironi** [comment]
We can now go ahead using `staging`.

**Pull request merged**